src/common/position.rs

```diff
@@ -83,6 +83,10 @@ impl Position {
         )
     }
 
+    pub fn invisible() -> Self {
```

Comment:
Perhaps we should get rid of `Default` implementation for invisible, or make this the default.
Confusing to have both.
Conceptually better to not have default since that doesn't make sense, `invisible` is more descriptive.